jujugui/static/gui/src/app/views/utils.js

```diff
@@ -1822,31 +1846,40 @@ YUI.add('juju-view-utils', function(Y) {
   */
   utils.generateServiceName = function(
     charmName, charmId, services, ignoreSelf) {
-    var charmCount = 0;
     // Loop through each service and grab its charmid to see if it matches.
-    services.each((service) => {
+    var highestCounter = Math.max.apply(Math, services.map((service) => {
       // Remove the version number from the charm id as we don't care about the
       // version when comparing charms.
       var serviceCharmId = utils._extractCharmName(service.get('charm'));
       var providedCharmId = utils._extractCharmName(charmId);
       // If we have a matching charm then increase the count.
       if (serviceCharmId === providedCharmId) {
-        charmCount += 1;
+        // Remove the service name and dash from the name to get the counter.
+        var counter = service.get('name').replace(
```

Comment:
This is because of two things. Firstly, if we just split on a hyphen we would get 'gui' from 'juju-gui' so by replacing the name we know we will only get the counter. Secondly, the first service name won't include the hypen. e.g. first service will be "wordpress" so if we were to remove "wordpress-" we'd be left with 'wordpress' which is not what we want. I'll extend the comment.